settings.gradle.kts

```diff
@@ -121,6 +121,13 @@ include(":runners:core-java")
 include(":runners:direct-java")
 include(":runners:extensions-java:metrics")
 /* Begin Flink Runner related settings */
+/* When updating these versions, please make sure that the following files are updated as well:
+  * FLINK_VERSIONS in .github/actions/setup-default-test-properties/test-properties.json
+  * flink_versions in sdks/go/examples/wasm/README.md
+  * PUBLISHED_FLINK_VERSIONS in sdks/python/apache_beam/options/pipeline_options.py
+  * PUBLISHED_FLINK_VERSIONS in sdks/typescript/src/apache_beam/runners/flink.ts
+  * verify versions in website/www/site/content/en/documentation/runners/flink.md
```

Comment:
I saw this update, but would someone outside google know _how_ to update this? Currently this references version 1.15, what is the rule?

Reply:
Yeah, I didn't know this in advance and only used external stuff when I was updating it, so anyone should be able to. Its based on compatibility with a version here - https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-2.1 - so next would be 1.17.0 for https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-2.2

I guess https://github.com/apache/beam/blob/5ff55b1caf109ce1da44500598dfaf4e85ef1caa/sdks/python/apache_beam/runners/interactive/interactive_beam.py#L410 isn't totally clear. Basically, this is just another option for spinning up a flink cluster if you don't have one ready to go.

Reply:
Could this be made part of Dataproc release cycle?

Reply:
I don't think so? I think that typically, Beam's Flink support is behind Dataproc's; plus, we don't necessarily need to be on the latest Flink supported by Dataproc.

I think it makes more sense to include this as a Beam task when we deprecate a Flink version used by the interactive runner. I don't think this is much toil, is there a reason to try to offload it?

Reply:
I find it generally non-intuitive for OSS project to have such dependencies on proprietary solutions. I understand the historical reasons, though. On the other hand, can non-googler be confident enough to change these settings? When can we change to 2.2? I really don't know, which is why it would feel more 'stable' to offload this.

Reply:
> On the other hand, can non-googler be confident enough to change these settings? When can we change to 2.2? I really don't know, which is why it would feel more 'stable' to offload this.

To be clear, I don't have any particular special insights here. I'd never used Dataproc before I made my change and don't really have any connections to the team. I chose 2.1 because that's what the rest of our project uses for testing and I was trying to keep my changes scoped small for the bugfix, not because there's a particular reason 2.2 wouldn't work. We could add some comments recommending that approach if you'd like?

> I find it generally non-intuitive for OSS project to have such dependencies on proprietary solutions.

I don't think this is particularly different than having a Dataflow runner or supporting GCP/Azure/AWS filesystems (or other arbitrary IOs); there's no reason we couldn't support other clouds or local setups (in fact we already do, we just don't spin up a cluster automatically for those setups yet). The only reason is that nobody from the community has had enough time or know-how to contribute an implementation (which is also fine). I'd actually love it if we had an AWS Managed Flink offering here, and we could very easily version that differently than Dataproc.

The dataproc piece only impacts people who want to use dataproc and have GCP-specific pipeline options set, which seems reasonable to me.

Reply:
> I really don't know, which is why it would feel more 'stable' to offload this.

Also, I don't think we have anybody engaged with the Beam community who is working on Dataproc (or has worked on it in the past AFAIK), so there's not really anyone who we could ask here.

Reply:
OK. :)